src/lava/lib/optimization/apps/scheduler/problems.py

```diff
@@ -0,0 +1,310 @@
+        self.graph = ntx.Graph()
+        self.generate_valid_nodes()
+        self.generate_edges_from_constraints()
+        self.rescale_adjacency()
+
+    def generate_valid_nodes(self):
+        node_id = 0
+        if self.agent_attrs is None:
+            self.agent_attrs = np.reshape(self.agent_ids,
+                                          (len(self.agent_ids), 1))
+        agent_id_attr_map = dict(zip(self.agent_ids, self.agent_attrs))
+        if self.task_attrs is None:
+            self.task_attrs = (
+                np.tile(np.reshape(self.task_ids,
+                                   (len(self.task_ids), 1)), (1, 2)))
+        task_id_attr_map = dict(zip(self.task_ids, self.task_attrs))
+        for aid, a_attr in agent_id_attr_map.items():
+            for tid, t_attr in task_id_attr_map.items():
+                if self.is_node_valid(aid, tid):
+                    self.graph.add_node(node_id,
+                                        agent_id=aid,
+                                        task_id=tid,
+                                        agent_attr=a_attr,
+                                        task_attr=t_attr)
+                    node_id += 1
+
+    def generate_edges_from_constraints(self):
+        num_nodes = len(self.graph.nodes)
+        self.adjacency = (
+            np.zeros((num_nodes, num_nodes), dtype=int))
+        for n1 in self.graph.nodes:
+            for n2 in self.graph.nodes:
+                not_same = n1 != n2
+                is_conflict = self.is_edge_conflicting(n1, n2)
+                if not_same and is_conflict:
+                    self.graph.add_edge(n1, n2)
+                    self.adjacency[n1, n2] = 1
+
+    def rescale_adjacency(self):
+        """ Scale the adjacency matrix weights for QUBO solver. """
+        self.adjacency = np.triu(self.adjacency)
+        self.adjacency += self.adjacency.T - 2 * np.diag(
+            self.adjacency.diagonal())
+
+
+class SatelliteScheduleProblem(SchedulingProblem):
+    """
+    SatelliteScheduleProblem is a synthetic scheduling problem in which a
+    number of vehicles must be assigned to view as many requests in a
+    2-dimensional plane as possible. Each vehicle moves horizontally across
+    the plane, has minimum and maximum view angle, and has a maximum rotation
+    rate (i.e. the rate at which the vehicle can reorient vertically from one
+    target to the next).
+
+    The problem is represented as an infeasibility graph and can be solved by
+    finding the Maximum Independent Set.
+    """
+
+    def __init__(
+            self,
+            num_satellites: int = 6,
+            view_height: float = 0.25,
+            view_coords: Optional[np.ndarray] = None,
+            num_requests: int = 48,
+            requests: Optional[np.ndarray] = None,
+            turn_rate: float = 2,
+            solution_criteria: float = 0.99,
+            seed: int = 42,
+    ):
+        """ Create a SatelliteScheduleProblem.
```

Comment:
I think we add the parameters in the docstring above the __init__, don't we?

Reply:
Fixed.